### PR TITLE
[CoreML EP] Add support for LeakyReLU activation layers

### PR DESCRIPTION
**Pull request by ShukantPal**
## Description

Implements support for LeakyReLU in ActivationOpBuilder for CoreML's EP.

### Motivation and Context

This speeds up inference on macOS significantly for models using LeakyReLU.

### CODE REVIEW

**Comment by ShukantPal** [comment]
@microsoft-github-policy-service agree

File: onnxruntime/core/providers/coreml/builders/impl/activation_op_builder.cc [hunk 1]
```diff
@@ -95,6 +95,19 @@ Status ActivationOpBuilder::AddToModelBuilderImpl(ModelBuilder& model_builder,
   } else if (op_type == "PRelu") {
     auto* prelu = layer->mutable_activation()->mutable_prelu();
     ORT_RETURN_IF_ERROR(AddPReluWeight(model_builder, node, logger, *prelu));
+  } else if (op_type == "LeakyRelu") {
+    const auto& attributes = node.GetAttributes();
+    auto alpha_attr = attributes.find("alpha");
+
+    if (alpha_attr != attributes.end()) {
+      auto alpha = (*alpha_attr).second.f();
+      auto *leaky_relu = layer->mutable_activation()->mutable_leakyrelu();
```
Comment (YUNQIUGUO) [inline onnxruntime/core/providers/coreml/builders/impl/activation_op_builder.cc:104]: you can use the NodeAttributeHelper to get the `alpha` attribute, as what we do in other places, for example:

https://github.com/microsoft/onnxruntime/blob/786f8b98f7a8ed6165212bed48c7c01fc86ea624/onnxruntime/core/providers/coreml/builders/impl/LRN_op_builder.cc#L41-L43

File: onnxruntime/core/providers/coreml/builders/impl/activation_op_builder.cc [hunk 2]
```diff
@@ -95,6 +95,19 @@ Status ActivationOpBuilder::AddToModelBuilderImpl(ModelBuilder& model_builder,
   } else if (op_type == "PRelu") {
     auto* prelu = layer->mutable_activation()->mutable_prelu();
     ORT_RETURN_IF_ERROR(AddPReluWeight(model_builder, node, logger, *prelu));
+  } else if (op_type == "LeakyRelu") {
+    const auto& attributes = node.GetAttributes();
```
Comment (YUNQIUGUO) [inline onnxruntime/core/providers/coreml/builders/impl/activation_op_builder.cc:99]:
```suggestion
    NodeAttrHelper helper(node);
    const auto alpha = helper.Get("alpha", 0.01f);
```
using 0.01 as the default value for alpha - LeakyRelu

**Review by YUNQIUGUO — COMMENTED** [review]
thanks for the contribution!

**Comment by ShukantPal** [comment]
Thanks for the review @YUNQIUGUO! − I implemented your suggestion.

**Comment by natke** [comment]
If you can, please share the models/scenarios you are using with this op. Thanks!

**Review by YUNQIUGUO — APPROVED** [review]

**Pull request merged**